### PR TITLE
Fix #679, Add Markdown version of OSAL Configuration Guide

### LINKED ISSUE #679: Convert doc/OSAL-Configuration-guide.doc to markdown
**Is your feature request related to a problem? Please describe.**
Doc and PDF not GitHub friendly

**Describe the solution you'd like**
Convert to markdown and remove doc/pdf

**Describe alternatives you've considered**
None

**Additional context**
None

**Requester Info**
Jacob Hageman - NASA/GSFC

### PR DESCRIPTION
**Pull request by ArielSAdamsNASA**
**Describe the contribution**
Pandoc was used to convert the OSAL Configuration Guide to a markdown file. During the conversion, issues were presented such as broken tables. Tables were fixed and block codes were added where appropriate. 

Fix #679

**Additional context**
Please provide feedback on whether the markdown file is acceptable in terms of formatting. 

**Contributor Info - All information REQUIRED for consideration of pull request**
Ariel Adams, ASRC Federal

### CODE REVIEW

**Comment by skliper** [comment]
@astrogeco @ArielSAdamsNASA - what's the status on this?  It would help to move it forward...

**Comment by skliper** [comment]
Also delete the other versions... markdown only preferred.

**Comment by ArielSAdamsNASA** [comment]
> @astrogeco @ArielSAdamsNASA - what's the status on this? It would help to move it forward...

@skliper Waiting on @acudmore for approval/feedback. I sent him an email this morning.

**Comment by skliper** [comment]
Related to #384, the issue to update the contents.  A format check here should be sufficient (and will simplify fixing #384).

File: doc/OSAL-Configuration-Guide-Markdown.md
Comment (astrogeco) [inline doc/OSAL-Configuration-Guide-Markdown.md:1]: Remove front matter and have the first section be the Introduction

File: doc/OSAL-Configuration-Guide-Markdown.md
Comment (astrogeco) [inline doc/OSAL-Configuration-Guide-Markdown.md:27]: Maybe move these to an appendix of sorts? I'm not sure we need this when the document is versioned using git

File: doc/OSAL-Configuration-Guide-Markdown.md
Comment (astrogeco) [inline doc/OSAL-Configuration-Guide-Markdown.md:140]: Use markdown "#" syntax for all headings

File: doc/OSAL-Configuration-Guide-Markdown.md
Comment (astrogeco) [inline doc/OSAL-Configuration-Guide-Markdown.md:287]: Use Markdown syntax and get rid of horizontal rule "-----"

File: doc/OSAL-Configuration-Guide-Markdown.md
Comment (astrogeco) [inline doc/OSAL-Configuration-Guide-Markdown.md:615]: Make this a fourth-level heading "####"

File: doc/OSAL-Configuration-Guide-Markdown.md
Comment (astrogeco) [inline doc/OSAL-Configuration-Guide-Markdown.md:663]: Use a code block for this

File: doc/OSAL-Configuration-Guide-Markdown.md
Comment (astrogeco) [inline doc/OSAL-Configuration-Guide-Markdown.md:683]: Use Fourth-level headings "####"

**Review by astrogeco — CHANGES_REQUESTED** [review]
Mainly format cleanups. It might be worthwhile to do a find-and-replace on bold and italic syntax "***" which is sometimes used for subheadings in this document. Similarly horizontal rules "===" are also being used in a subheading fashion. 

It might be worth passing this through a markdown linter like remark or prettier; there are Atom and VSCode plugins for to help with that.

**Review by astrogeco — CHANGES_REQUESTED** [review]
Remove "markdown" from the file name and use [`git rm` ](https://www.atlassian.com/git/tutorials/undoing-changes/git-rm) to remove the pdf and doc versions of the guide.

File: doc/OSAL-Configuration-Guide.md
Comment (ArielSAdamsNASA) [inline doc/OSAL-Configuration-Guide.md:1]: Removed front matter so the document begins with the Introduction.

File: doc/OSAL-Configuration-Guide.md
Comment (ArielSAdamsNASA) [inline doc/OSAL-Configuration-Guide.md:861]: Placed Revision History at the bottom of the document for now.

**Comment by ArielSAdamsNASA** [comment]
I removed the pdf and doc versions of the guide and updated the markdown version.

For the markdown version:
- Removed the front matter, so the Introduction is the first section 
- Updated headers to use the markdown syntax # and removed horizontal rule for headers
- Updated table headers to use fourth-level headers 
- Use a code block for the section starting with **\$ sparc-rtems4.10-gdb tasking-example.nxe**
- Moved Revision History to the bottom of the page for now

File: doc/OSAL-Configuration-Guide.md
Comment (acudmore) [inline doc/OSAL-Configuration-Guide.md:45]: This table does not look like it is rendering correctly, at least when viewed on GitHub.
This is what it looks like when I view it.
osal-markdown

**Review by acudmore — APPROVED** [review]
The conversion is good, there are a couple of table format issues ( in the glossary for example ). But this conversion and removing the old documents is the first step, I think the doc needs to be overhauled in content after this conversion. Much of the technical info is out of date.

File: doc/OSAL-Configuration-Guide.md
Comment (ArielSAdamsNASA) [reply doc/OSAL-Configuration-Guide.md]: Updated several tables in terms of formatting. Should be all set to be marked as CCB-READY?

**Comment by astrogeco** [comment]
Looks good! I would do an interactive rebase and squash the table fix into the original md commit. Also edit the commit message to use the format "Fix #XYZ, COMMIT_DESCRIPTION"

**Comment by astrogeco** [comment]
[CCB 2020-12-16](https://github.com/nasa/cFS/wiki/Community-CCB-Agenda:-December-16-2020) APPROVED

- Separate issue for content

**Review by astrogeco — APPROVED** [review]

**Pull request merged**
